api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto

```diff
@@ -310,4 +310,97 @@ message TcpProxy {
   //   :ref:`core.v3.ProxyProtocolConfig.pass_through_tlvs <envoy_v3_api_field_config.core.v3.ProxyProtocolConfig.pass_through_tlvs>`
   //   for details.
   repeated config.core.v3.TlvEntry proxy_protocol_tlvs = 19;
+
+  // Configuration for controlling when to establish the upstream connection.
+  //
+  // When not specified, the filter uses the default ``IMMEDIATE`` mode for backward compatibility.
+  //
+  // .. attention::
+  //   Server-first protocols (e.g., SMTP, MySQL, POP3) require ``IMMEDIATE`` mode. Data-triggered
+  //   modes will cause connection timeouts as the client waits for the server greeting.
+  UpstreamConnectionEstablishmentMode upstream_connection_establishment_mode = 21;
```

Comment:
The name is so long. Would only `UpstreamConnectMode upstream_connect_mode` be better?